api/src/main/java/org/infinispan/api/reactive/query/ContinuousQueryRequestBuilder.java

```diff
@@ -0,0 +1,68 @@
+package org.infinispan.api.reactive.query;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Builder class to build {@link QueryRequest} that can be used for Continuous Queries
+ *
+ * Different parameters can be specified:
+ * <u>
+ *    <li>Ickle Query: Mandatory query string</li>
+ *    <li>Param: Parameters for the query/li>
+ *    <li>All: Get created, updated and removed entries</li>
+ *    <li>Created: Get created entries</li>
+ *    <li>Updated: Get updated entries</li>
+ *    <li>Deleted: Get deleted entries</li>
+ * </u>
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 10.0
+ */
+public final class ContinuousQueryRequestBuilder {
+   private String ickleQuery;
+   private Map<String, Object> params;
+   private boolean created;
```

Comment:
what about using an enum instead?

Reply:
considering this as a refactoring if more type of CQ event types are added. Today there are only 3 and you can say I want created and removed ones for example